### PR TITLE
Liberasurecode

### PR DESCRIPTION
**Pull request by ehudkaldor**
adding liberasurecode, a library for Erasure Code

### CODE REVIEW

**Comment by Ikke** [comment]
Previous PR: #4684

File: testing/liberasurecode/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	./configure \
+		--build=$CBUILD \
+		--host=$CHOST \
+		--prefix=/usr \
+		--sysconfdir=/etc \
+		--mandir=/usr/share/man \
+		--localstatedir=/var
+	make
+}
+
```
Comment (TBK) [inline testing/liberasurecode/APKBUILD]:
```suggestion

check() {
    make test
}

```

**Comment by TBK** [comment]
Please squash the commits and rename it as well as the PR to: `testing/liberasurecode: new aport`

**Comment by ehudkaldor** [comment]
@TBK ,
@maxice8 (who commented the same on the previous PR),

when i remove all the `builddir` references, my build fails as it does not get into the right dir to `make`.

```
# Contributor: Ehud Kaldor <ehud@unfairfunction.org>
# Maintainer: Ehud Kaldor <ehud@unfairfunction.org>
pkgname=liberasurecode
pkgver=1.6.0
pkgrel=0
pkgdesc="Erasure Code API library written in C with pluggable Erasure Code backends"
url="https://github.com/openstack/liberasurecode"
arch="all"
license="BSD-2-Clause"
makedepends="autoconf automake libtool zlib-dev"
subpackages="$pkgname-dev"
source="$pkgname-$pkgver.tar.gz::https://github.com/openstack/liberasurecode/archive/$pkgver.tar.gz"
#builddir="$srcdir"/$pkgname-$pkgver

prepare() {
#	cd $builddir
	./autogen.sh
}

build() {
	./configure \
		--build=$CBUILD \
		--host=$CHOST \
		--prefix=/usr \
		--sysconfdir=/etc \
		--mandir=/usr/share/man \
		--localstatedir=/var
	make
}

check() {
#	cd $builddir
  make test
}

package() {
#	cd $builddir
	make DESTDIR="$pkgdir" install
}
sha512sums="168fb75f2e329e79d6cbbc5d97e7a60963f742d9cce6e492a3c845ef77acc204b00a34292e847ae0c165177fa38a8563de2428d01f8f47364af4fb1e8f4d9517  liberasurecode-1.6.0.tar.gz"

```

result:

```
>>> liberasurecode: Installing for build: build-base autoconf automake libtool zlib-dev
(1/14) Installing m4 (1.4.18-r1)
(2/14) Installing libbz2 (1.0.6-r6)
(3/14) Installing perl (5.26.3-r0)
(4/14) Installing autoconf (2.69-r2)
(5/14) Installing automake (1.16.1-r0)
(6/14) Installing ncurses-terminfo-base (6.1_p20180818-r1)
(7/14) Installing ncurses-terminfo (6.1_p20180818-r1)
(8/14) Installing ncurses-libs (6.1_p20180818-r1)
(9/14) Installing readline (7.0.003-r0)
(10/14) Installing bash (4.4.19-r1)
Executing bash-4.4.19-r1.post-install
(11/14) Installing libltdl (2.4.6-r5)
(12/14) Installing libtool (2.4.6-r5)
(13/14) Installing zlib-dev (1.2.11-r1)
(14/14) Installing .makedepends-liberasurecode (0)
Executing busybox-1.28.4-r2.trigger
OK: 234 MiB in 66 packages
>>> liberasurecode: Cleaning temporary build dirs...
>>> liberasurecode: Fetching liberasurecode-1.6.0.tar.gz::https://github.com/openstack/liberasurecode/archive/1.6.0.tar.gz
>>> liberasurecode: Fetching liberasurecode-1.6.0.tar.gz::https://github.com/openstack/liberasurecode/archive/1.6.0.tar.gz
>>> liberasurecode: Checking sha512sums...
liberasurecode-1.6.0.tar.gz: OK
>>> liberasurecode: Unpacking /var/cache/distfiles/liberasurecode-1.6.0.tar.gz...
/usr/bin/abuild: line 1: ./autogen.sh: not found
>>> ERROR: liberasurecode: all failed
>>> liberasurecode: Uninstalling dependencies...
(1/14) Purging .makedepends-liberasurecode (0)

```

if i add `builddir`, it succeeds. what am i doing wrong?

**Comment by maxice8** [comment]
Starting with v3.3.0 of abuild the prepare() build() check() and package() functions automatically change the working directory to the value of builddir. I guess your abuild version is on stable ?

**Comment by ehudkaldor** [comment]
true enough. i was using Alpine 3.8, which has `abuild` 3.2. update to Alpine 3.9, and all worked.

**Comment by ehudkaldor** [comment]
auto-closed as I deleted the branch on github. will open a new PR.

**Comment by Ikke** [comment]
@ehudkaldor Note that you can just push new commits (or rebase and force push) to update the pull request, it's not necessary to delete the branch.

**Comment by ehudkaldor** [comment]
Yup. I'm learning on the fly...

**Pull request closed without merging**